atlasdb-config/src/main/java/com/palantir/atlasdb/factory/timestamp/FreshTimestampSupplierAdapter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory.timestamp;
+
+import java.util.function.Supplier;
+
+import com.google.common.base.Preconditions;
+import com.palantir.exception.NotInitializedException;
+import com.palantir.timestamp.TimestampService;
+
+public class FreshTimestampSupplierAdapter implements Supplier<Long> {
```

Comment:
Now we aren't planning on using memoizeWithExpiration I'd switch back.
I actually initially used primitive LongSupplier, but switched to this because we were using MWE. I could write an adapter but that would box/unbox the longs anyway, and I didn't really fancy copying the ExpiringMemoizingSupplier from Guava and writing a primitive version